linera-execution/tests/contract_runtime_apis.rs

```diff
@@ -952,3 +954,81 @@ async fn test_query_service(authorized_apps: Option<Vec<()>>) -> Result<(), Exec
 
     Ok(())
 }
+
+/// Tests the contract system API to make HTTP requests.
+#[cfg(feature = "unstable-oracles")] // # TODO: Remove once #3524 lands
+#[test_case(vec![()] => matches Ok(_); "when authorized")]
+#[test_case(vec![] => matches Err(ExecutionError::UnauthorizedApplication(_)); "when unauthorized")]
+#[test_log::test(tokio::test)]
+async fn test_perform_http_request(authorized_apps: Vec<()>) -> Result<(), ExecutionError> {
+    let mut view = SystemExecutionState {
+        description: Some(ChainDescription::Root(0)),
+        ownership: ChainOwnership::default(),
+        balance: Amount::ONE,
+        balances: BTreeMap::new(),
+        ..SystemExecutionState::default()
+    }
+    .into_view()
+    .await;
+
+    let contract_blob = TransferTestEndpoint::sender_application_contract_blob();
+    let service_blob = TransferTestEndpoint::sender_application_service_blob();
+    let contract_blob_id = contract_blob.id();
+    let service_blob_id = service_blob.id();
+
+    let application_description = TransferTestEndpoint::sender_application_description();
+    let application_description_blob = Blob::new_application_description(&application_description);
+    let app_desc_blob_id = application_description_blob.id();
+
+    let (application_id, application) = view
+        .register_mock_application_with(application_description, contract_blob, service_blob)
+        .await
+        .expect("should register mock application");
+
+    let make_http_requests = authorized_apps
+        .into_iter()
+        .map(|_| application_id)
```

Comment:
Optional minor nit: we could use `()` here directly to make it fail to compile if we change the test parameter but forget to update it here.

```suggestion
        .map(|()| application_id)
```